client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,135 @@
+import { gql, useMutation } from '@apollo/client';
+import React, { FunctionComponent, useEffect,  useState } from 'react';
+import  Form  from 'react-bootstrap/Form';
+import moment from 'moment';
+import Request from '../../data/types/request';
+import RequestForm from "../organisms/RequestForm";
+import  Table  from 'react-bootstrap/Table';
+
+interface Props {
+    requests: Request[];
+    onChangeNumRequests?: (num : number) => void;
+}
+
+const ClientRequestTable: FunctionComponent<Props> = (props: Props) => {
+    const [ requestSelectedForEditing, setRequestSelectedForEditing ] = useState("");
+    const headingList = ['Fulfilled', 'Request Group', 'Request Type', 'Quantity', 'Date Requested', ''];
+    const [requests, setRequests] = useState<Request[]>([]);
+    
+    const updateRequest = gql` 
+    mutation updateRequest($request: RequestInput){
+        updateRequest(request: $request){
+          id
+          success
+          message
+        }
+      }
+    `;
+
+    const softDeleteRequest = gql`
+    mutation deleteRequest($id: ID) {
+        softDeleteRequest(id: $id) {
+            id
+            success
+            message
+        }
+    }`
+    
+    function orderRequests(undeletedReq : Request[]) {
+        const filterByFulfilled = (requests : Request[], keepFulfilled : boolean) => {
+            requests = requests.filter( request => {
+                    if (request !== undefined){
+                        if (request.fulfilled === keepFulfilled){
+                            return request;
+                        }
+                    }
+                }
+            );
+            return requests;
+        }
+        const unfulfilledReq = filterByFulfilled(undeletedReq, false);
+        const fulfilledRequests = filterByFulfilled(undeletedReq, true);
+
+        const compareDateCreated = (a : Request, b : Request) => {
+            return (a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf()); 
+        }
+        unfulfilledReq.sort(compareDateCreated);
+        fulfilledRequests.sort(compareDateCreated);
+    
+        const sortedRequests : Request[] = unfulfilledReq!.concat(fulfilledRequests!) as Request[];
+        setRequests(sortedRequests);
+        props.onChangeNumRequests!(sortedRequests.length);
+    }
+
+    useEffect(() => {
+        const undeletedReq : Request[] = props.requests.filter((request)=> request.deleted === false);
+        orderRequests(undeletedReq);
+      }, [props.requests]);
+      
+
+    const [mutateDeleteRequest] = useMutation(softDeleteRequest);
+    const [mutateRequest] = useMutation(updateRequest);
```

Comment:
Use these hooks at the beginning of the function (put them at lines 37)
We generally put mutation hooks after their query definitions